### PR TITLE
Airtable schema migration

### PR DESCRIPTION
**Pull request by dangalan**
Start moving to the [new Airtable base](https://airtable.com/appF8rJZ3XAkzWOO9/tbl2Cq3znN4NBsyAL/viwip94i6FfyYsc0r?blocks=hide). There's a lot more refactoring + reorganization that needs to be done but this PR is already big so I'll break that up into future ones.

**Note: I didn't test very thoroughly locally because we're gonna be rewriting with Gracie and Mike's new design so this some old functionality might be fucked.**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "editor.codeActionsOnSave": {
+    "source.sortImports": true
+  },
+  "editor.defaultFormatter": "esbenp.prettier-vscode",
+  "editor.formatOnSave": true
+}
```
Comment (dangalan) [inline .vscode/settings.json:7]: Copied this from work lol - if you're using VSCode it'll auto-format so we don't have to run the formatter manually as much.

File: api/clients/airtable.ts [hunk 2]
```diff
@@ -15,30 +15,29 @@ export class AirtableClient {
     this.base = new Airtable({ apiKey: apiKey }).base(baseId)
   }
 
-  async find<T>(tableName: string, id: string): Promise<T> {
-    const record = await this.base.table(tableName).find(id)
-    return record.fields as unknown as T
+  async find<T extends FieldSet>(tableName: string, id: string): Promise<T> {
```
Comment (dangalan) [inline api/clients/airtable.ts:18]: Without extending `FieldSet`, I was getting some type error in VSCode even though the type check script wasn't picking it up 🤷🏼‍♂️ 
Doesn't seem any harm in doing it.

File: api/services.ts [hunk 3]
```diff
@@ -2,7 +2,7 @@ import { AirtableClient, AirtableCreateResponse, AxiosClient } from './clients'
 import {
   Address,
   CreateServicesListRequest,
-  Service,
+  Resource,
```
Comment (dangalan) [inline api/services.ts:5]: We should files and components with "Resource" too but we can do that in a separate PR since this one already has a lot of changes.

File: components/ServiceItem.tsx [hunk 4]
```diff
@@ -12,19 +12,18 @@ import {
...
   Heading,
   HStack,
-  Link,
   LinkBox,
   LinkOverlay,
   Stack,
   Text,
 } from '@chakra-ui/react'
-import { Address, Service } from 'models'
+import { AddressWithLabel, Resource } from 'models'
```
Comment (dangalan) [inline components/ServiceItem.tsx:20]: Addresses in the new Airtable schema aren't in a separate table with names attached so created a new type that links addresses with labels.

File: components/ServiceItem.tsx [hunk 5]
```diff
@@ -152,12 +146,12 @@ export const ServiceItem = ({
...
       )}
-      {service.phoneNumbers && (
+      {service.phone && (
         <LinkBox>
           <HStack spacing="8px">
             <PhoneIcon />
-            <LinkOverlay href={`tel:${service.phoneNumbers[0]}`}>
-              {service.phoneNumbers[0]}
+            <LinkOverlay href={`tel:${service.phone}`}>
+              {service.phone}
```
Comment (dangalan) [inline components/ServiceItem.tsx:154]: Resources only have one phone number in the new schema.

File: hooks/useServicesList.ts [hunk 6]
```diff
@@ -43,56 +41,23 @@ export const useServiceList = (listId: string): ServiceListHandler => {
...
-    isLoadingServices ? [] : (baseServices as Service[]),
-    ['name', 'description'],
-    'taxonomyString'
+    isLoadingServices ? [] : (baseServices as Resource[]),
+    ['title', 'details'],
+    'needs'
   )
 
-  const addressIdToServiceName: Record<string, string> = {}
-  const addressQueryOptions =
```
Comment (dangalan) [inline hooks/useServicesList.ts:58]: Address fields are on resources in the new schema so we no longer need a separate query for them.

File: models/services.ts [hunk 7]
```diff
@@ -1,43 +1,49 @@
 export type Status = 'Draft' | 'Published'
 
-export interface Service {
+export type ResourceStatus =
```
Comment (dangalan) [inline models/services.ts:3]: Switching to types over interfaces because we can't index interfaces with strings (i.e. `Interface[stringField]`) without adding [an index signature](https://www.typescriptlang.org/docs/handbook/2/objects.html#index-signatures).

File: api/services.ts [hunk 8]
```diff
@@ -59,40 +48,34 @@ export const postServicesList = async (
...
 // Airtable functions
 
-export const findAddress = (id: string): Promise<Address> => {
-  return ServicesAirtableClient.find<Address>('physical_addresses', id)
-}
-
-export const findService = (id: string): Promise<Service> => {
-  return ServicesAirtableClient.find<Service>('services', id)
+export const findService = (id: string): Promise<Resource> => {
+  return ServicesAirtableClient.find<Resource>('Resources', id)
```
Comment (dangalan) [inline api/services.ts:61]: The strings based in here are Airtable names - switching them to match the tables in new Airtable base.

We should prob make the env variables and not hardcoded at some point but that's for another PR.

File: models/needs.ts [hunk 9]
```diff
@@ -0,0 +1,16 @@
+export type Need = {
```
Comment (dangalan) [inline models/needs.ts:1]: Model for the `Ref - Need` table

**Review by vronnieli — APPROVED** [review]
